dcos_test_utils/gce.py

```diff
@@ -0,0 +1,174 @@
+        elif not os.path.isfile(credentials_path):
+            raise Exception(credentials_path + ' is not a valid google credentials path.')
+        else:
+            os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = credentials_path
+
+        credentials = GoogleCredentials.get_application_default()
+        self.compute = discovery.build('compute', 'v1', credentials=credentials)
+        self.deployment_manager = discovery.build('deploymentmanager', 'v2', credentials=credentials)
+
+        with open(credentials_path, 'r') as creds:
+            self.project_id = json.load(creds)['project_id']
+
+    def deploy_instances(self, config):
+        body = {
+            'name': config['deployment_name'],
+            'target': {
+                'config': {
+                    'content': config['template_content']
+                }
+            }
+        }
+
+        request = self.deployment_manager.deployments().insert(project=self.project_id, body=body)
+        response = request.execute()
+        return response
+
+    def list_deployment_instances(self, deployment):
+        request = self.compute.instances().list(project=self.project_id, zone=deployment.zone)
+        instance_info_list = []
+
+        while request is not None:
+            response = request.execute()
+            for instance_info in response['items']:
+                instance_info_list.append(instance_info)
+
+            request = self.deployment_manager.resources().list_next(previous_request=request, previous_response=response)
+
+        return instance_info_list
+
+    def get_deployment_info(self, deployment):
+        request = self.deployment_manager.deployments().get(project=self.project_id, deployment=deployment.name)
+        response = request.execute()
+        return response
+
+    def delete_deployment(self, deployment):
+        request = self.deployment_manager.deployments().delete(project=self.project_id, deployment=deployment.name)
+        response = request.execute()
+        return response
+
+    def set_metadata(self, deployment, body, instance_name):
+        request = self.compute.instances().setMetadata(project=self.project_id, zone=deployment.zone,
+                                                       instance=instance_name, body=body)
+        response = request.execute()
+        return response
+
+
+class Instance:
+    def __init__(self, gce_wrapper, name, priv_ip, pub_ip):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.priv_ip = priv_ip
+        self.pub_ip = pub_ip
+
+    def get_private_public_ip(self):
+        return {
+            'private_ip': self.priv_ip,
+            'public_ip': self.pub_ip}
+
+
+class Deployment:
+    def __init__(self, gce_wrapper, name, zone):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.zone = zone
+        self.instances = []
+        self.fingerprint = None
+
+    @catch_http_exceptions
```

Comment:
Forget this comment. This was a good use of a decorator. Two changes however:
- decorator should use probably use log.exception instead of just raising the exception
- catch_http_exceptions should be sprinkled on the GceWrapper because the Deployment object should not be thinking about what the GceWrapper is doing with HTTP requests